lib/polyfill/orientation.js

```diff
@@ -23,21 +24,47 @@ shaka.polyfill.Orientation = class {
    * @export
    */
   static install() {
-    if (screen.orientation) {
+    if (screen.orientation && screen.orientation.unlock) {
       // Not needed.
+      shaka.log.info('Using native screen.orientation');
       return;
     }
 
-    // There is no way to check to see if the 'orientationchange' event exists
-    // on window, which could theoretically lead to this making a
-    // screen.orientation object that doesn't actually work.
-    // However, it looks like all platforms that support the deprecated
-    // window.orientation feature also support that event.
-    if (window.orientation != undefined) {
+    if (screen.orientation != undefined) {
+      // There are some platforms where screen.orientation is defined but
+      // incomplete (e.g. Safari).
+      // Install a very simple polyfill in that case.
+      shaka.polyfill.Orientation.installBasedOnScreenMethods_();
+    } else if (window.orientation != undefined) {
+      // There is no way to check to see if the 'orientationchange' event exists
+      // on window, which could theoretically lead to this making a
+      // screen.orientation object that doesn't actually work.
+      // However, it looks like all platforms that support the deprecated
+      // window.orientation feature also support that event.
       shaka.polyfill.Orientation.installBasedOnWindowMethods_();
     }
   }
 
+  /**
+   * Modifies screen.orientation to add no-ops for missing methods.
+   * Meant for cases where screen.orientation is defined, but has missing
+   * methods that cannot be properly polyfilled.
+   * @private
+   */
+  static installBasedOnScreenMethods_() {
+    if (screen.orientation.lock === undefined) {
+      screen.orientation.lock = (orientation) => {
+        shaka.log.info('screen.orientation.lock is a no-op');
+        return Promise.resolve();
+      };
+    }
+    if (screen.orientation.unlock === undefined) {
+      screen.orientation.unlock = () => {
+        shaka.log.info('screen.orientation.unlock is a no-op');
```

Comment:
Same question here.

Reply:
This one improves things because, on desktop Safari, neither `lock` nor `unlock` are implemented. So we don't need to unlock the screen, because we never locked it in the first place; we just need to prevent the method call from erroring, so that we can move on to the part of the code that calls `document.exitFullscreen`.
I guess technically this approach, where each methods is filled with a no-op independently, would lead to a weird result if the browser has `lock` implemented but doesn't have `unlock` implemented. But I would hope that no browser provides an API for locking the screen without providing an API for unlocking it, that sounds like an accident waiting to happen.

Reply:
I had the impression from this PR and the comments on it that only unlock() was missing on Safari.  Thanks for clarifying.